pkg/storage/sqlstorage/store.go

```diff
@@ -12,33 +13,41 @@ const (
 	SQLCustomFuncMetaCompare = "meta_compare"
 )
 
-type Store struct {
-	schema  Schema
-	onClose func(ctx context.Context) error
+type API struct {
+	schema   Schema
+	executor executor
 }
 
-func (s *Store) Schema() Schema {
+func (s *API) Schema() Schema {
 	return s.schema
 }
 
-func (s *Store) error(err error) error {
+func (s *API) error(err error) error {
 	if err == nil {
 		return nil
 	}
 	return errorFromFlavor(Flavor(s.schema.Flavor()), err)
 }
 
-func NewStore(schema Schema, onClose func(ctx context.Context) error) (*Store, error) {
-	return &Store{
-		schema:  schema,
-		onClose: onClose,
-	}, nil
+func NewAPI(schema Schema, executor executor) *API {
+	return &API{
+		schema:   schema,
+		executor: executor,
+	}
 }
 
-func (s *Store) Name() string {
+func (s *API) Name() string {
 	return s.schema.Name()
 }
 
+var _ storage.API = &API{}
```

Comment:
What is this line for?

Reply:
Again, a compile time assertion to ensure *API implements storage.API interface.
Not really required in this case since the code will not compile if the structure does not implements the interface.
In the case of the json.Marshaler, it would compile.

Even if not strictly required in this case, it helps to see directy than *API implement storage.API.

Also, Intellij/Goland will propose you to implements the methods automatically ;)

Reply:
Alright thanks :)